browser/metric_mapping.go

```diff
@@ -0,0 +1,43 @@
+package browser
+
+import (
+	"fmt"
+
+	"github.com/grafana/xk6-browser/common"
+)
+
+// mapMetric to the JS module.
+func mapMetric(vu moduleVU, cm *common.ExportedMetric) (mapping, error) {
+	rt := vu.VU.Runtime()
+
+	// We're setting up the function in the Sobek context that will be reused
+	// for this VU.
+	_, err := rt.RunString(`
+	function _k6BrowserURLGroupingTest(pattern, url) {
+		let r = pattern;
+		if (typeof pattern === 'string') {
+			r = new RegExp(pattern);
+		}
+		return r.test(url);
+	}`)
```

Comment:
Unfortunately there are differences between the Golang and JS (ECMA) regex engines (as there are with [all regex engines](https://en.wikipedia.org/wiki/Comparison_of_regular_expression_engines)). Since the user will be writing the regex in the JS test script, I believe it is better to stay consistent and use the JS (Sobek's) regex engine instead of Go's implementation to avoid confusion.

Reply:
👍